### PR TITLE
Align stories, works and images and set an empty default state

### LINKED ISSUE #8862: Search: Stories, finalise layout and functionalities
This ticket can only be finalised once Pagination and Sort tickets are finished.

Ensure it has these components and they work and follow the correct layout
- (Pagination)
- (# results)
- (Sort)

To do:
- Replace static data with real data on the FE (dependent on #8825, might be done in that branch already(?))
- Change types for some to be optional. This could allow for a default state (we don't need results if there is no query). We could also deal with that differently, but needs to be decided upon.
- Integrate default state (no results)

### PR DESCRIPTION
**Pull request by rcantin-w**
## Who is this for?
Devs and new search

## What is it doing for them?
- Aligns all three search pages in terms of how to render different states
- Images and Works don't display default results when the query is empty anymore
- We don't use `SearchNoResults` anymore when the query is empty, it's solely for when a query returns no results again.
- `hasFilters` for `SearchNoResults` now looks at all filters (not sure why it was only looking at specific ones - maybe this change wasn't necessary?)
- Moved each pages' H1 to `SearchPageLayout`
- Added `<link rel="next/prev" ... />` to `/search/works`. Probably should be added to stories too but only did it quickly here as part of the alignment, I think the rest depends on Pagination for Prismic content working.

Closes #8862 , #8863 , #8864

### CODE REVIEW

File: catalogue/webapp/components/SearchPageLayout/SearchPageLayout.tsx [hunk 1]
```diff
@@ -151,6 +151,14 @@ const SearchLayout: FunctionComponent = ({ children }) => {
             return false;
           }}
         />
+        <h1 className="visually-hidden">
```
Comment (rcantin-w) [inline catalogue/webapp/components/SearchPageLayout/SearchPageLayout.tsx:154]: Tbh this content is subject to change, I just wanted to move it out of the individual files as we were rendering a different component should the query string be empty.

File: catalogue/webapp/pages/search/images.tsx [hunk 2]
```diff
@@ -63,21 +65,32 @@ const BottomPaginationWrapper = styled(PaginationWrapper)`
...
   const { setLink } = useContext(SearchContext);
   useEffect(() => {
     const link = toLink({ ...imagesRouteProps }, 'images_search_context');
     setLink(link);
   }, [imagesRouteProps]);
 
+  // If there is no query, return an empty page
+  if (!queryString) {
+    return (
+      <Space
```
Comment (rcantin-w) [inline catalogue/webapp/pages/search/images.tsx:81]: Not a huge fan of the empty component but we want spacing and I'm thinking this is still subject to change? Open to suggestions!

File: catalogue/webapp/pages/search/works.tsx [hunk 3]
```diff
@@ -198,7 +268,7 @@ export const getServerSideProps: GetServerSideProps<Props | AppErrorProps> =
     const _queryType = getCookie('_queryType') as string | undefined;
 
     const worksApiProps = {
-      ...props,
+      ...params,
```
Comment (rcantin-w) [inline catalogue/webapp/pages/search/works.tsx:271]: Changed it to be the same variable name as `/search/images` uses 🤷‍♀️

File: catalogue/webapp/components/SearchNoResults/SearchNoResults.tsx [hunk 4]
```diff
@@ -31,18 +31,16 @@ const SearchNoResults: FunctionComponent<Props> = ({
...
-                  </>
-                )}
-              </Copy>
-            )}
+            <Copy textColor={textColor}>
+              We couldn&rsquo;t find anything that matched{' '}
+              <QuerySpan>{query}</QuerySpan>
+              {hasFilters && (
+                <>
+                  <span> with the filters you have selected</span>. Please try
```
Comment (alexwlchan) [inline catalogue/webapp/components/SearchNoResults/SearchNoResults.tsx:39]: I think we can get rid of this `<span>`.

File: catalogue/webapp/components/SearchPageLayout/SearchPageLayout.tsx [hunk 5]
```diff
@@ -151,6 +151,14 @@ const SearchLayout: FunctionComponent = ({ children }) => {
...
+          {/* Uppercases the first letter */}
+          {`${
+            currentSearchCategory[0].toUpperCase() +
+            currentSearchCategory.substring(1)
+          } search page`}
```
Comment (alexwlchan) [inline catalogue/webapp/components/SearchPageLayout/SearchPageLayout.tsx:159]: You can use https://github.com/wellcomecollection/wellcomecollection.org/blob/main/common/utils/grammar.ts#L1-L6

File: catalogue/webapp/pages/search/images.tsx [hunk 6]
```diff
@@ -138,36 +154,38 @@ const ImagesSearchPage: NextPageWithLayout<Props> = ({
...
+                queryParams: Object.keys(query).map(p => p),
+              })}
+            />
+          ) : (
+            <Wrapper>
+              <div className="container">
+                <PaginationWrapper>
+                  <span>{`${images.totalResults} result${
+                    images.totalResults > 1 ? 's' : ''
+                  }`}</span>
```
Comment (alexwlchan) [inline catalogue/webapp/pages/search/images.tsx:173]: Let's look at styling this consistently with TotalResultsCopy in stories.tsx, and possibly pushing down the pluralisation copy into the component as well.

File: catalogue/webapp/pages/search/images.tsx [hunk 7]
```diff
@@ -177,12 +195,29 @@ ImagesSearchPage.getLayout = getSearchLayout;
...
+    // Stop here if no query has been entered
+    if (!params.query) {
+      return {
+        props: removeUndefinedProps({
+          imagesRouteProps: params,
+          serverData,
+          query,
+          pageview: {
+            name: 'images',
+            properties: { totalResults: 0 },
```
Comment (alexwlchan) [inline catalogue/webapp/pages/search/images.tsx:215]: "I entered a search query and I got no results" and "I didn't enter a search query" are two different events, can we track them differently?

File: catalogue/webapp/pages/search/works.tsx [hunk 8]
```diff
@@ -96,11 +112,34 @@ export const CatalogueSearchPage: NextPageWithLayout<Props> = ({
 
   return (
     <>
-      <h1 className="visually-hidden">Works Search Page</h1>
+      <Head>
+        {works?.prevPage && (
```
Comment (alexwlchan) [inline catalogue/webapp/pages/search/works.tsx:116]: We talked about how this is for Google spider-ing reasons and needs a bigger conversation; let's create a ticket to track those discussions.

File: catalogue/webapp/utils/search.ts [hunk 9]
```diff
@@ -56,3 +56,20 @@ export const getPrismicSortValue = ({
   // this is currently what we set as default for Stories
   return 'title_ASC';
 };
+
+// FILTERS
+/**
+ * Compare filter options to query parameters
+ * This is used for SearchNoResult's rendered copy
+ * @param {string[]} filters - Available filter options
+ * @param {string[]} queryParams - URL query parameters
```
Comment (alexwlchan) [inline catalogue/webapp/utils/search.ts:65]: > Tell us if the user has applied any filters

**Review by alexwlchan — APPROVED** [review]

File: catalogue/webapp/pages/search/images.tsx [hunk 10]
```diff
@@ -63,21 +65,32 @@ const BottomPaginationWrapper = styled(PaginationWrapper)`
...
 
   const { setLink } = useContext(SearchContext);
   useEffect(() => {
     const link = toLink({ ...imagesRouteProps }, 'images_search_context');
     setLink(link);
   }, [imagesRouteProps]);
 
+  // If there is no query, return an empty page
+  if (!queryString) {
+    return (
```
[the remark this answers is not in the document]
Comment (rcantin-w) [reply catalogue/webapp/pages/search/images.tsx:80]: I should review this with @jtweed and confirm it's what we want as a default state or if we'd like to make it look slightly different/add messaging.

Screenshot 2022-11-29 at 10 07 31

File: catalogue/webapp/pages/search/works.tsx (hunk 8 above)
Comment (rcantin-w) [reply catalogue/webapp/pages/search/works.tsx:116]: https://github.com/wellcomecollection/wellcomecollection.org/issues/8936

Did a bit more research and documented it in there

File: catalogue/webapp/pages/search/images.tsx (hunk 7 above)
Comment (rcantin-w) [reply catalogue/webapp/pages/search/images.tsx:214]: [Started a thread](https://wellcome.slack.com/archives/C02ANCYL90E/p1669717732492449) for @harrisonpim but he's not in today so I'll treat this separately.

**Pull request merged**